cmd/client/search.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/Masterminds/squirrel"
+	pgUtils "github.com/monstercat/golib/db/postgres"
+	"github.com/monstercat/gologx/logxhost"
+	"github.com/olekukonko/tablewriter"
+)
+
+func showSearch(name string, args []string) error {
+	var postgres string
+	var service string
+	var matcher string
+	var limit int
+	var datebefore string
+	var dateafter string
+	var orderby string
+
+	limitDefault := 50
+
+	set := flag.NewFlagSet(name, flag.ExitOnError)
+	set.StringVar(&postgres, "postgres", "", "Postgres database")
+	set.StringVar(&service, "service", "", "Defines a service to be filtered")
+	set.StringVar(&matcher, "matcher", "", "Defines a matcher to be filtered. Searches machine and name")
+	set.IntVar(&limit, "limit", limitDefault, "Defines the limit of results in the list")
+	set.StringVar(&datebefore, "datebefore", "", "Filters before a specific date time")
+	set.StringVar(&dateafter, "dateafter", "", "Filters after a specific date time")
+	set.StringVar(&orderby, "orderby", "", "Orders list by a defined column")
+
+	if err := set.Parse(args); err != nil {
+		return err
+	}
+
+	if service == "" {
+		fmt.Println(styleDanger("service flag is required"))
+		os.Exit(1)
+	}
+
+	dateRegex := regexp.MustCompile(`\d{4}-\d{2}-\d{2}`)
+	if valid := dateRegex.MatchString(datebefore); datebefore != "" && !valid {
```

Comment:
This is not a valid date check? 9999-99-99 would pass. Also, dateBefore != "" is irrelevant. "" doesn't match the regex =)

Reply:
Can also probably use https://golang.org/pkg/time/#Parse instead